### ISSUE TITLE
Proposal for team to purchase another RoboRio

### ISSUE THREAD

**Issue by RajGandhi** [FRCTeam3206/Programming#9]
Programming team to identify cost, benefits and plan for use.  Add to this issue.

Plan to present (a couple minutes) to team leads / mentors at next available opportunity.

**Comment by Daniel-Thalman** [comment]
I feel like i am missing the point of another rio

**Comment by RajGandhi** [comment]
For software development, wouldn't it be good (ideal) for each developer to have their own RoboRio, with one or more motors and a full set of sensors.  Weren't there issues last season with limited RoboRios for software developing.  

Don't be immediately be dissuaded by the $500 cost.

**Comment by ghost** [comment]
We had mentioned the idea of several training stations that programmers
could use rather than all crowding around one robot or the CRio from years
past.  A table-top setup with Rio, power, motor + encoder, servo,
ultrasonic, and camera would be fantastic and likely easier to get new
coders trained more efficiently.  Perhaps a minimal demonstration checklist
of required skills would be useful to transition coders from table top
training to bot support for drive team to new hardware development.  Just
thinking.

-Rob

**Comment by steeve-o** [comment]
I like Rob's thinking. A working table top system might also enable us to
get ahead of the build curve. Last year there was a lot of competition for
time on the bot with the mechanical team. If we had a table top system, we
could potentially do much of our design and debugging without the bot.

Do we know at this point if we will be working towards 2 bots / 1
competition or 1 bot / 2 competitions for next year? If we chose the
latter, we would already have a roborio available for use as a table
bot(although having 2 wouldn't be bad either).

**Comment by steeve-o** [comment]
I just read Raj's earlier comment. It would be really nice to have multiple roborio work stations available to programmers. I feel the programming team kind of self limits the number of students that it can take in just based off of availability of the roborio. Adding additional work stations (even with partial sensor sets), would help keep members involved, and would help spread the workload (instead of the programmer who best knows the code always being the one working on it, in the spirit of efficiency because there is one workstation). Assuming we stick with java, we could assign programmers to write individual libraries/classes etc, which they could live test on a robo rio. These scripts would then be easily integrateable into the main program on the bot. I think the way classes and libraries are organized and called in java is an advantage of using it over labview, where the integration (at least as students used it) of individual programs into the main vi seemed to be a bit messy and problematic. 
In summary, multiple roborios would help us achieve the following team structure:
One programmer (architect) to write the main program for the bot, and the junior programmers can be used to write needed functions/classes/libraries etc.

**Comment by jessicamd** [comment]
I like the idea of work stations as well. I believe there has been a push for increased funding and the idea of having both 2 bots and 2 competitions is a goal this year, so there might not be a RoboRio to spare for Programming. If there going to be a focus on obtaining these work stations, then having specific funding from companies for this might be good so that they know they are investing in programming learning. I also feel that having a table top board would be nice for Systems members to see how the input of different sensors and motor connections work.

**Comment by Daniel-Thalman** [comment]
This would also encourage programming members to get into systems and vis versa add ing to its value

**Comment by jessicamd** [comment]
Absolutely, it opens up possibilities for cooperation and new projects that will have added value in that they would be more similar to working on the actual robot.
